extension/index.googleFunction.js

```diff
@@ -7,31 +7,57 @@ const logger = utils.getLogger()
 
 exports.extensionTrigger = async (request, response) => {
   try {
+    const obj = request?.body?.resource?.obj
+    if (!obj) {
+      return utils.sendGoogleFunctionResponse({
+        response,
+        statusCode: 400,
+        body: {
+          errors: [
+            {
+              code: 'BadRequest',
+              message: 'Invalid body payload.',
+            },
+          ],
+        },
+      })
+    }
     const authToken = auth.getAuthorizationRequestHeader(request)
-    const { obj } = request.body.resource
     const paymentResult = await paymentHandler.handlePayment(obj, authToken)
     if (paymentResult.success) {
-      response.status(200).send({
-        actions: paymentResult.data ? paymentResult.data.actions : [],
-      })
-    } else {
-      response.status(400).send({
-        errors: paymentResult.data ? paymentResult.data.errors : undefined,
+      return utils.sendGoogleFunctionResponse({
+        response,
+        statusCode: 200,
+        body: {
+          actions: paymentResult.data ? paymentResult.data.actions : [],
+        },
       })
     }
+    return utils.sendGoogleFunctionResponse({
+      response,
+      statusCode: 400,
+      body: {
+        errors: paymentResult.data ? paymentResult.data.errors : undefined,
+      },
+    })
   } catch (err) {
-    const errorMessage = `Unexpected error: ${JSON.stringify(
+    const errorMessage = `Unexpected error: ${err.message}`
+    const errorStackTrace = `Unexpected error: ${JSON.stringify(
```

Comment:
I would always try to include payment ID from the `obj` in the logged stack traces.

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/630/commits/60231a2c7d727dace55bde76e966374a30fb59e1